userspace/libsinsp/threadinfo.cpp

```diff
@@ -155,9 +155,9 @@ void sinsp_threadinfo::init()
 	m_exe_ino_ctime_duration_clone_ts = 0;
 	m_exe_ino_ctime_duration_pidns_start = 0;
 
-	memset(&m_user, 0, sizeof(scap_userinfo));
+	m_user = nullptr;
+	m_loginuser = nullptr;
```

Comment:
Maybe we can directly initialize the pointers depending on the compile time flag here so we don't have to do a runtime checks on `set_user` and `set_loginuser`. WDYT?
```suggestion
#ifdef SINSP_USER_DETAILS
        m_user = std::make_shared<user_details>();
        m_loginuser = std::make_shared<user_details>();
#else
	m_user = nullptr;
	m_loginuser = nullptr;
#endif
```

Reply:
Yeah, why not -- this would make it even closer to the original implementation.